charts/portworx/templates/storage-cluster.yaml

```diff
@@ -289,21 +292,64 @@ spec:
   {{- end}}
   {{- end }}
 
-  {{- if (or (and (.Values.monitoring) (eq .Values.monitoring true)) (and (.Values.telemetry) (eq .Values.telemetry true) )) }}
+  {{- if or (and (.Values.monitoring.telemetry) (eq .Values.monitoring.telemetry true))
+  (and $prometheus.enabled (eq $prometheus.enabled true))
+  (and .Values.grafana (eq .Values.grafana true))
+  }}
   monitoring:
-    {{- if (and (.Values.monitoring) (eq .Values.monitoring true)) }}
+    {{- if (and ( $prometheus.enabled) (eq $prometheus.enabled true)) }}
     prometheus:
```

Comment:
we should split conditions here for "promethues" attribute seperately, like

{if prometheus is enabled or export metrics is enabled}
promethues:
{if prometheus is enabled}
  enabled: true
{if exportMetrics is enabled}
  exportMetrics: true

 
since we have cases with external prometheus where prometheus is disabled but exportMetrics is enabled

Reply:
@nikita-bhatia I have modified conditions as mentioned.